### ISSUE TITLE
Error while pulling image -- lookup index.docker.io on [::1]:53

### ISSUE THREAD

**Issue by aharpervc** [moby/moby#22398]
**Output of `docker version`:**

```
Client:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Wed Apr 27 00:34:20 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Wed Apr 27 00:34:20 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.11.1
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 0
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host
Kernel Version: 4.4.8-boot2docker
Operating System: Boot2Docker 1.11.1 (TCL 7.0); HEAD : 7954f54 - Wed Apr 27 16:36:45 UTC 2016
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 995 MiB
Name: rake-compiler-machine
ID: ELVY:Z7XK:A4LJ:BEBA:MPOX:5ORT:MUOS:UCZA:NYVM:W4OY:S7ZL:QUYC
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug mode (client): false
Debug mode (server): true
 File Descriptors: 12
 Goroutines: 29
 System Time: 2016-04-28T15:29:34.6086957Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Labels:
 provider=hyperv
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
- running the commands from Windows 10 x64 to a newly created docker machine in hyperv

**Steps to reproduce the issue:**
1. create a new hyperv docker machine & run the command to set the docker environment
2. run `docker run -t -i ubuntu /bin/bash`
3. observe output

**Describe the results you received:**

There's an error about accessing the docker repository:

```
$ docker run -t -i ubuntu /bin/bash
Unable to find image 'ubuntu:latest' locally
Pulling repository docker.io/library/ubuntu
C:\Users\--\docker.exe: Error while pulling image: Get https://index.docker.io/v1/repositories/library/ubuntu/images: dial tcp: lookup index.docker.io on [::1]:53: read udp [::1]:55313->[::1]:53: read: connection refused.
See 'C:\Users\--\docker.exe run --help'.
```

**Describe the results you expected:**

I expected docker to connect to the registry, grab the appropriate image, etc

**Additional information you deem important (e.g. issue happens only occasionally):**

Seems to happen every time. It appears to be trying to access index.docker.io on localhost, `lookup index.docker.io on [::1]:53` but I don't have anything in my hosts file or any other weird dns config, and no proxies. Ping times out (public icmp disabled?), but the host lookup seems to be correct:

```
$ ping index.docker.io

Pinging us-east-1-elbio-rm5bon1qaeo4-623296237.us-east-1.elb.amazonaws.com [52.73.94.64] with 32 bytes of data:
Request timed out.
```

Maybe this is an ipv6 thing? Can I force docker to use ipv4 somehow?

**Comment by aaronlehmann** [comment]
I think this may be an issue with the virtual machine setup. If you ssh into the Linux VM, can you do DNS lookups at all, for example using the `host` command?

**Comment by HackToday** [comment]
seems some network issue, did you use ipv6? or any special setting in your network?

**Comment by carlosvin** [comment]
I had the same issue, I fixed it disabling ipv6. 

IHMO it is a workaround, do you foresee to support ipv6?

**Comment by aharpervc** [comment]
I don't have any special ipv6 configuration set up, but i'd prefer to not make a machine configuration change if possible. Is there a way to just configure docker to not use ipv6?

**Comment by aharpervc** [comment]
I disabled ipv6 on all adapters (no luck) then [disabled ipv6 system-wide](https://support.microsoft.com/en-us/kb/929852) (no luck). Then I nuked the existing internal virtual switch, and used another adapter, and that worked. So definitely something weird with my networking config.

However, that doesn't really explain why docker is using localhost to connect to index.docker.io

**Comment by szantopeter** [comment]
an uninstall/reinstall fixed the problem (also uninstalled docker toolbox)

**Comment by SwanJumelle** [comment]
I had the exact same problem, my DNS settings was not properly configured

**Comment by jtougas** [comment]
I had a similar problem with Docker for windows when running `docker run hello-world`
```
Unable to find image 'hello-world:latest' locally
docker: Error response from daemon: Get https://registry-1.docker.io/v2/: dial tcp: lookup registry-1.docker.io on [::1]:53: read udp [::1]:47504->[::1]:53: read: connection refused.
See 'docker run --help'.
```
This was running an outdated client on WSL on a machine that had some dubious VPN clients installed which I figure might have played a role... In any case the same command from powershell was spitting out the same error message.

I changed the dns server to fixed 8.8.8.8 in the control panel thingy and that resolved the problem. The strange part is that shortly after I set it back to automatic and things work fine now.

![image](https://user-images.githubusercontent.com/7550618/37987446-304ac730-31cc-11e8-8553-58b7efeda61c.png)
